spec/rspec/rails/matchers/active_job_spec.rb

```diff
@@ -98,6 +98,31 @@ def self.name; "LoggingJob"; end
       expect { }.not_to have_enqueued_job
     end
 
+    context "when job is retried" do
```

Comment:
What confuses me is that this spec is passing with the original implementation, too.
Can you write one that would fail with the original, and would pass with yours/mine?